asserts/ifacedecls_test.go

```diff
@@ -956,6 +985,79 @@ func (s *plugSlotRulesSuite) TestCompilePlugRuleConnectionConstraintsDeviceScope
 	}
 }
 
+func (s *plugSlotRulesSuite) TestCompilePlugRuleConnectionConstraintsSideArityConstraints(c *C) {
+	m, err := asserts.ParseHeaders([]byte(`iface:
+  allow-auto-connection: true`))
+	c.Assert(err, IsNil)
+
+	rule, err := asserts.CompilePlugRule("iface", m["iface"].(map[string]interface{}))
+	c.Assert(err, IsNil)
+
+	// defaults
+	c.Check(rule.AllowAutoConnection[0].SlotsPerPlug, Equals, asserts.SideArityConstraint{N: 1})
+	c.Check(rule.AllowAutoConnection[0].PlugsPerSlot.Any(), Equals, true)
+
+	c.Check(rule.AllowConnection[0].SlotsPerPlug.Any(), Equals, true)
+	c.Check(rule.AllowConnection[0].PlugsPerSlot.Any(), Equals, true)
+
+	// allow-connection => *
+	allowConnTests := []string{
+		`iface:
+  allow-connection:
+    slots-per-plug: 1
+    plugs-per-slot: 2`,
+		`iface:
+  allow-connection:
+    slots-per-plug: *
+    plugs-per-slot: 1`,
+		`iface:
+  allow-connection:
+    slots-per-plug: 2
+    plugs-per-slot: *`,
+	}
+
+	for _, t := range allowConnTests {
+		m, err = asserts.ParseHeaders([]byte(t))
+		c.Assert(err, IsNil)
+
+		rule, err = asserts.CompilePlugRule("iface", m["iface"].(map[string]interface{}))
+		c.Assert(err, IsNil)
+
```

Comment:
The test shows how 1, 2 and * all get clamped to *.
I would be good to say why this is the case here. It's not at all clear from reading the test itself.

Reply:
I'll add a comment.